armi/reactor/components/complexShapes.py

```diff
@@ -91,6 +91,62 @@ def getCircleInnerDiameter(self, Tc=None, cold=False):
             return 0.0
 
 
+class HexHoledCircle(basicShapes.Circle):
+    """Circle with n uniform hexagon holes hollowed out of it."""
+
+    THERMAL_EXPANSION_DIMS = {"od", "holeOP"}
+
+    pDefs = componentParameters.getHexHoledCircleParameterDefinitions()
+
+    def __init__(
+        self,
+        name,
+        material,
+        Tinput,
+        Thot,
+        od,
+        holeOP,
+        mult=1.0,
+        modArea=None,
+        isotopics=None,
+        mergeWith=None,
+        components=None,
+    ):
+        ShapedComponent.__init__(
+            self,
+            name,
+            material,
+            Tinput,
+            Thot,
+            isotopics=isotopics,
+            mergeWith=mergeWith,
+            components=components,
+        )
+        self._linkAndStoreDimensions(
+            components, od=od, holeOP=holeOP, mult=mult, modArea=modArea
+        )
+
+    def getComponentArea(self, cold=False):
+        r"""Computes the area for the hexagon with n number of circular holes in cm^2."""
+        od = self.getDimension("od", cold=cold)
+        holeOP = self.getDimension("holeOP", cold=cold)
+        mult = self.getDimension("mult")
+        hexArea = math.sqrt(3.0) / 2.0 * (holeOP ** 2)
+        circularArea = math.pi * ((od / 2.0) ** 2)
+        area = mult * (circularArea - hexArea)
+        return area
+
+    def getHexInnerPitch(self, Tc=None, cold=False):
+        """
+        For the special case of only one single hole, returns the
+        diameter of that hole.
+
+        For any other case, returns 0.0 because an "circle inner diameter" becomes
+        undefined.
+        """
+        return 0.0
```

Comment:
I think that you are intending to implement the `getCircleInnerDiameter()` method here? We have a requirement (for thermal expansion purposes) to implement a method on all component shapes called `getCircleInnerDiameter`, which basically returns the diameter of a circle that would be inscribed by the inner dimension.

For this case, I think you should (1) return 0.0 if there are multiple holes, or (2) return the diameter of the circle inscribed in the hexagonal hole like [this](https://media.geeksforgeeks.org/wp-content/uploads/Untitled-143-300x199.png).

If I'm interpreting incorrectly, and you actually _do_ want a method called `getHexInnerPitch()`, then the method above doesn't appear to do what you describe it to do. And, we should still implement a `getCircleInnerDiameter()` method as well.

Reply:
I think the way I changed it meets the intent of (2), but let me know if I still didn't get it quite right! Thank you!